src/OpenApi/src/Schemas/OpenApiJsonSchema.Helpers.cs

```diff
@@ -185,37 +186,34 @@ public static void ReadProperty(ref Utf8JsonReader reader, string propertyName,
                     var types = ReadList<string>(ref reader);
                     foreach (var type in types ?? [])
                     {
-                        // JSON Schema represents nullable types using an array consisting of
-                        // the target type and "null". Since OpenAPI Schema does not support
-                        // representing types within an array we must check for the "null" type
-                        // and map it to OpenAPI's `nullable` property for OpenAPI v3.
-                        if (type == "null")
+                        if (schema.Type is not null)
                         {
-                            schema.Nullable = true;
+                            schema.Type |= Enum.Parse<JsonSchemaType>(type, ignoreCase: true);
                         }
                         else
                         {
-                            schema.Type = type;
+                            schema.Type = Enum.Parse<JsonSchemaType>(type, ignoreCase: true);
```

Comment:
Why ignoreCase here and elsewhere when parsing type? Where does this value of type come from that it might not not be the correct case (and that would be okay)?

Reply:
When System.Text.Json gives us the schema type, it is serialized as a string in all lowercase (`object`, `null`, `array`, etc.) The `JsonSchemaType` enum provided by the OpenAPI uses title casing. `Enum.Parse` is case-sensitive by default so `object` won't map to `Object` unless we use this flag.

Reply:
> The JsonSchemaType enum provided by the OpenAPI uses title casing.

Isn't that a bug?

Reply:
No, as most enums use title-case (or I guess Pascal scale) for the type names. STJ maps them to an all lower-case identifier [here](https://source.dot.net/#System.Text.Json/System/Text/Json/Schema/JsonSchema.cs,58bdb337d5a99cb8,references)) which is why we get the inconsistency in casing between what `Enum.ToString()` while generate and what the schema will produce.